penrose-web/src/engine/Evaluator.ts

```diff
@@ -332,19 +323,141 @@ export const evalExpr = (
     }
     case "List": {
       const argVals = evalExprs(e.contents, trans, varyingVars, optDebugInfo);
+
+      // The below code makes a type assumption about the whole list, based on the first elements
+      // Is there a better way to implement parametric lists in typescript? 
```

Comment:
For now, one way is to figure out the element type in `Lists`s in Haskell land (i.e. style compiler when checking blocks) and encode them with this info. In general, the new TS parser should probably have a more specific type for lists.

Reply:
Oh yeah, encoding the expected type of a (singly-typed) list in the backend would be helpful in the frontend. But once we move to Typescript then we should figure out how to model lists properly.